tessellate_ipu/linalg/tile_linalg_hessenberg.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) 2023 Graphcore Ltd. All rights reserved.
+import os
+from typing import Any, Tuple
+
+import jax.lax
+import numpy as np
+from jax.core import ShapedArray
+
+from tessellate_ipu import TileShardedArray, create_ipu_tile_primitive, tile_map, tile_put_replicated, tile_put_sharded
+from tessellate_ipu.core.tile_interpreter_vertex_utils import make_ipu_vector1d_worker_offsets
+
+Array = Any
+
+
+def get_qr_vertex_gp_filename() -> str:
+    return os.path.join(os.path.dirname(__file__), "../core", "vertex", "tile_qr_vertex.cpp")
+
+
+dot_product1d_p = create_ipu_tile_primitive(
+    "dot_product1d",
+    "DotProduct1dVertex",
+    inputs=["x", "y"],
+    outputs={"partials": ShapedArray((12,), dtype=np.float32)},
+    constants={
+        "worker_offsets": lambda inavals, *_: make_ipu_vector1d_worker_offsets(
+            inavals[0].size, vector_size=2, num_workers=6, wdtype=np.uint16
+        )
+    },
+    # tmp_space=ShapedArray((12,), dtype=np.float32),
+    gp_filename=get_qr_vertex_gp_filename(),
+    perf_estimate=1000,
+)
+
+"""Vertex computing QR correction vector.
+"""
+qr_correction_vector_p = create_ipu_tile_primitive(
+    "qr_correction_vector",
+    "QRCorrectionVectorVertex",
+    inputs=["Rcol", "sdiag"],
+    outputs={"v": 0, "vrescale": ShapedArray((1,), dtype=np.float32)},
+    gp_filename=get_qr_vertex_gp_filename(),
+    perf_estimate=1000,
+)
+
+"""Vertex QR HouseHolder performing row inplace update: x -= scale1[0] * scale2[0] * v
+"""
+qr_householder_row_update_p = create_ipu_tile_primitive(
```

Comment:
I think we could import these 3 primitives instances from `tile_linalg_qr`, to avoid copy?